### PR TITLE
refactor Runtime to use F consistently

### PR DESCRIPTION
**Pull request by odeac**
## Overview
Use F consistently in Runtime and the classes that use it

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
https://rchain.atlassian.net/browse/RHOL-1050

### Complete this checklist before you submit the PR
- [ ] This PR contains no more than 200 lines of code, excluding test code.
- [ ] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [ ] If this PR adds a new feature, this PR includes tests related to this feature.
- [ ] You assigned one person to review this PR

### If you are not a member of the core development team, please confirm:
- [ ] You signed the commit. Merging requires a signature. Please see the [developer wiki](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain) for instructions.
- [ ] Your GitHub account is also an account with [Travis CI](https://travis-ci.org). Unit tests will not run on your PR unless you have an account with Travis. Merge requires passed unit tests.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala [hunk 1]
```diff
@@ -1,96 +1,89 @@
...
+      //TODO: Is better error handling needed here?
+//      _ <- if (processedDeploy.status.isFailed)
+//            Sync[F].raiseError(new Exception("Processed deploy has failed"))
+//          else ()
+
+      returnChannel = Par().copy(exprs = Seq(Expr(GString(name))))
+      result        <- runtime.space.getData(returnChannel)
+
+      _ <- Sync[F].delay { runtimeContainer.put(runtime) }
+    } yield result.flatMap(_.a.pars)
```
Comment (KentShikama) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: So what happens if the call to captureResults fails? Does it throw a stacktrace and crash the node?

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala [hunk 2]
```diff
@@ -102,34 +95,32 @@ class RuntimeManager private (val emptyStateHash: ByteString, runtimeContainer:
...
     val runtime   = runtimeContainer.take()
     val blakeHash = Blake2b256Hash.fromByteArray(hash.toByteArray)
-    Try(runtime.space.reset(blakeHash).unsafeRunSync) match {
+    Try(runtime.space.reset(blakeHash)) match {
       case Success(_) => runtime
       case Failure(ex) =>
         runtimeContainer.put(runtime)
         throw ex
     }
   }
```
Comment (KentShikama) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: What is the difference between using .attempt and Try/Success/Failure?

File: node/src/main/scala/coop/rchain/node/NodeRuntime.scala [hunk 3]
```diff
@@ -269,6 +269,22 @@ class NodeRuntime private[node] (
...
+        EitherT.liftF[Task, CommError, A](Task.Par.unwrap(par.value).flatMap {
+        case Left(ex) => Task.raiseError(new Exception(ex.toString))
+        case Right(x) => Task.pure(x)
+      })
+    }
+
+    override def parallel: ~>[Effect, EffectPar]     = new ~>[Effect, EffectPar] {
+      override def apply[A](eff: Effect[A]): EffectPar[A] = ??? //TODO
+    }
+  }
```
Comment (KentShikama) [inline node/src/main/scala/coop/rchain/node/NodeRuntime.scala]: @rabbitonweb @ArturGajowy Can you double check this implementation?

Comment (odeac) [reply node/src/main/scala/coop/rchain/node/NodeRuntime.scala]: this is incomplete. I forgot to put the label WIP to this PR

Comment (odeac) [reply node/src/main/scala/coop/rchain/node/NodeRuntime.scala]: so I removed this implementation and I used Parallel.identity instead. Thanks @ArturGajowy for the pointer

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala (hunk 1 above)
Comment (odeac) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: Thje code was commented by mistake. It's fixed now

**Comment by odeac** [comment]
This approach is wrong as the tests are really difficult to fix. Retrying a new approach in a separate PR

**Pull request closed without merging**